src/mofaflex/_core/mofaflex.py

```diff
@@ -785,9 +857,35 @@ def _fit(self, data, preprocessor):
                 for scale in prior_scales.values():
                     scale[: self._n_dense_factors, :] = dense_scale
 
+        # guided factors
+        guiding_vars = GuidingVarsDataset(data, self._data_opts.guiding_vars_obs_keys)
         covariates = CovariatesDataset(data, self._data_opts.covariates_obs_key, self._data_opts.covariates_obsm_key)
+
+        # get unique categories for each guiding variable
+        for guiding_var_name, guiding_var_likelihood in self._model_opts.guiding_vars_likelihoods.items():
+            if guiding_var_likelihood == "Categorical":
+                guiding_var_categories = []
```

Comment:
wouldn't it be easier to make this a set and remove all the `unique` and `concatenate` stuff?